admin-notice/class-admin-notice.php

```diff
@@ -5,20 +5,35 @@
 class Admin_Notice {
 	public $message;
 	public $conditions;
+	public $dismiss_identifier;
+
+	const COOKIE_DELIMETER = '|';
 
 	/**
 	 * Create AdminNotice.
 	 *
 	 * @param string $message The text to be displayed.
 	 * @param Admin_Notice_Condition[] $conditions Conditions to pass in order to show the notice.
+	 * @param string $dismiss_identifier if provided the dismiss will become dissmisible
 	 */
-	public function __construct( string $message, array $conditions = [] ) {
+	public function __construct( string $message, array $conditions = [], string $dismiss_identifier = '' ) {
+		$position = strpos( $dismiss_identifier, self::COOKIE_DELIMETER );
+		if ( false !== $position ) {
+			\trigger_error( esc_html( "Admin Notice dissmiss identifier - $dismiss_identifier can't contain delimeter - " . self::COOKIE_DELIMETER ) );
+		}
+
 		$this->message = $message;
 		$this->conditions = $conditions;
+		$this->dismiss_identifier = $dismiss_identifier ? $dismiss_identifier : '';
```

Comment:
```suggestion
		$this->dismiss_identifier = $dismiss_identifier ?: '';
```

In a ternary condition, if the value being checked is the same as the truthy result, then since  PHP 5.3, the truthy result (lefthand operand) can be skipped.

Reply:
uuuh that is nice! However, this code is no longer needed it can be just a simple assignment. I forgot to remove it from the previous attempt.